File: todolist/src/main/java/study/yapp/todolist/exception/BusinessException.java
package study.yapp.todolist.exception;

import lombok.Getter;
import study.yapp.todolist.common.ResponseCode;

@Getter
public class BusinessException extends RuntimeException {
    private final ResponseCode responseCode;

    public BusinessException(ResponseCode responseCode) {
        this.responseCode = responseCode;
    }

    public BusinessException(String message, ResponseCode responseCode) {
        super(message);
        this.responseCode = responseCode;
    }

}

File: src/main/kotlin/com/example/studytodolist/common/exception/BusinessException.kt
package com.example.studytodolist.common.exception

import org.springframework.http.HttpStatus

class BusinessException(override val message: String, val code: Int, val status: HttpStatus): RuntimeException() {
    constructor(errorCode: ErrorCode) : this(errorCode.message, errorCode.code, errorCode.status)
}